File: java/day1/Part1.java
import java.util.*;
import java.nio.file.*;
import java.io.IOException;

public class Part1 {
    public static void main(String[] args) throws IOException {
        List<String> allLines = Files.readAllLines(Paths.get("part1.txt"));
        int sum = 0;
        List<Character> digits = Arrays.asList('0','1','2','3','4','5','6','7','8','9');
        for ( String line : allLines) {
            Integer firstDigit=null, lastDigit=null;
            Integer lineSum = 0;
            for ( char c : line.toCharArray() ) {
                if ( digits.contains(c) ) {
                    Integer parsed  = Integer.parseInt(new String(new char[]{c}));
                    if ( firstDigit == null ) {
                        firstDigit = parsed;
                    }
                    lastDigit = parsed;
                }
            }
            lineSum = (firstDigit * 10) + lastDigit;
            sum += lineSum;
        }
        System.out.println("Answer ::: " + sum);
    }
}

File: zig/day1/part1.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const fileContents = @embedFile("input.txt");
    for (fileContents) |line| {
        try stdout.print("{c}", .{line});
        //var firstDigit = undefined;
        //var lastDigit = undefined;
        for (line) |character| {
            try stdout.print("{c}", .{character});
        }
    }
    const hello = "hello";
    try stdout.print("{s}", .{hello});
}
